Mathlib/GroupTheory/ArchimedeanDensely.lean

```diff
@@ -68,6 +68,23 @@ lemma Subgroup.zpowers_eq_zpowers_iff {G : Type*} [CommGroup G] [LinearOrder G]
   refine hl.imp ?_ ?_ <;>
   simp +contextual
 
+lemma Int.addEquiv_eq_refl_or_neg (e : ℤ ≃+ ℤ) : e = .refl _ ∨ e = .neg _ := by
+  suffices e 1 = 1 ∨ - e 1 = 1 by
+    rw [neg_eq_iff_eq_neg] at this
+    simp_rw [← AddEquiv.toAddMonoidHom_injective.eq_iff]
+    exact this.imp (AddMonoidHom.ext_int ·) (AddMonoidHom.ext_int ·)
+  rw [← AddSubgroup.zmultiples_eq_zmultiples_iff]
+  simpa [e.surjective, eq_comm] using (AddMonoidHom.map_zmultiples (G := ℤ) (N := ℤ) (e : ℤ →+ ℤ) 1)
+
+instance : Unique (ℤ ≃+o ℤ) where
```

Comment:
Can you add the instance with `OrderDual` on the RHS too? Is there a more general `Subsingleton (ℤ ≃+o X)` version that's true?